### PR TITLE
added DTO and integrated with MockProcess controller

### PR DESCRIPTION
**Pull request by j0rdan0**
No business logic added yet, just added DTO and integrated this with MockProcess controller. Also used a formatter :P

### CODE REVIEW

File: mockOSApi.v2/Controllers/ProcessController.cs [hunk 1]
```diff
@@ -10,63 +10,70 @@
...
-    
-   
-   _handler.ChangePriority(prio.Prio,prio.Pid);
-   return Ok();
-
-}
+    }
+
+    [HttpPost]
+    public ActionResult CreateProcess(ProcessCreationDto process)
```
Comment (Horatzio) [inline mockOSApi.v2/Controllers/ProcessController.cs]: I like this. This is actually really good separation of concerns, to use different DTOs for different results

File: mockOSApi.v2/Data/DTO/ProcessDto.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
+using mockOSApi.Models;
+using mockOSApi.Repository;
+
+namespace mockOSApi.DTO;
+
+public class PriorityClass()
+{
+    public int Prio { get; set; }
```
Comment (Horatzio) [inline mockOSApi.v2/Data/DTO/ProcessDto.cs:9]: If the Priority is going to have specific values, you should use an Enum instead.

File: mockOSApi.v2/Data/DTO/ProcessDto.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
+using mockOSApi.Models;
+using mockOSApi.Repository;
+
+namespace mockOSApi.DTO;
+
+public class PriorityClass()
```
Comment (Horatzio) [inline mockOSApi.v2/Data/DTO/ProcessDto.cs:7]: So this class aggregates a priority to a Process?
I would suggest you just include the priority inside the process.
Is there another reason why you made it separately?

File: mockOSApi.v2/Data/DTO/ProcessDto.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    public string? Image { get; set; }
+
+    public string[]? Arguments { get; set; }
+
+    public int Priority { get; set; }
+
+}
+
+
+public interface IMapper
```
Comment (Horatzio) [inline mockOSApi.v2/Data/DTO/ProcessDto.cs]: This should be in a different file.

I imagine you started to find this mapping logic tedious.
Let me introduce you to [Automapper](https://docs.automapper.org/en/stable/Getting-started.html)

File: mockOSApi.v2/Models/OSObject.cs [hunk 5]
```diff
@@ -2,54 +2,62 @@
...
 
-public abstract class OSObject {
+public abstract class OSObject
+{
 
 
     /* Fields and properties
     *******************************
     */
     private int _handle;
```
Comment (Horatzio) [inline mockOSApi.v2/Models/OSObject.cs]: This is all equivalent to `public int Handle { get; set; }`
Unless you have some custom code involved, don't declare the private int property separately.

File: mockOSApi.v2/Models/OSObject.cs [hunk 6]
```diff
@@ -2,54 +2,62 @@
...
+    {
+        return this.HashCode;
 
+    }
+
+
+    private void SetHashCode()
+    {
+
+        byte[] source = new byte[14];
```
Comment (Horatzio) [inline mockOSApi.v2/Models/OSObject.cs]: So generally, the HashCode is based solely on the data inside the object.
Two objects that have the exact same values, should have the same HashCode.
If you want to make use of HashCode, start using [C# Records](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record).
They are a specialized class that implements by default a constructor, equals and HashCode functions

File: mockOSApi.v2/Models/Task/MockProcess.cs [hunk 7]
```diff
@@ -4,105 +4,119 @@
...
+        set => _state = value;
+    }
+
+    public string[]? Args
+    {
+        get => _arguments;
+        set => _arguments = value;
+    }
+
+    public int ExitCode
```
Comment (Horatzio) [inline mockOSApi.v2/Models/Task/MockProcess.cs:56]: Make this nullable, as you will only have the ExitCode after the process finishes

File: mockOSApi.v2/Models/Task/MockProcess.cs [hunk 8]
```diff
@@ -4,105 +4,119 @@
...
 public enum PROC_STATE
 {
-    Running, 
+    Running,
     Sleeping,
     Zombie
 };
 
-public enum DEFAULT_FD {
+public enum DEFAULT_FD
```
Comment (Horatzio) [inline mockOSApi.v2/Models/Task/MockProcess.cs]: The naming convetion for C# Enums is the same as other classes, but the members should be UPPERCASE_SNAKE_CASE.
Renamed this to DefaultFd

File: mockOSApi.v2/Repositories/ProcessRepository.cs [hunk 9]
```diff
@@ -3,51 +3,62 @@
...
     private readonly OSDbContext _dbContext;
-    public IEnumerable<MockProcess> GetAll() {
+    public IEnumerable<MockProcess> GetAll()
+    {
         return _dbContext.MockProcesses.ToList();
     }
-    public MockProcess? GetProcessByPid(int pid) {
+    public MockProcess? GetProcessByPid(int pid)
+    {
         return _dbContext.Find<MockProcess>(pid);
```
Comment (Horatzio) [inline mockOSApi.v2/Repositories/ProcessRepository.cs]:
```suggestion
        return _dbContext.MockProcesses.Where(p => p.pid == pid)
```
This is also natively supported by EF

File: mockOSApi.v2/Scripts/format.sh [hunk 10]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (Horatzio) [inline mockOSApi.v2/Scripts/format.sh:1]: Very nice!

File: mockOSApi.v2/Services/ProcessHandler.cs [hunk 11]
```diff
@@ -1,66 +1,82 @@
...
 using mockOSApi.Repository;
 using mockOSApi.Models;
+using mockOSApi.DTO;
 
 namespace mockOSApi.Services;
 
 
-public interface IProcessHandler {
-    public IEnumerable<MockProcess> AllProcesses { get; }
+public interface IProcessHandler
```
Comment (Horatzio) [inline mockOSApi.v2/Services/ProcessHandler.cs]: This should be IProcessService.
C# also uses the concept of 'Handlers', but for something totally different

File: mockOSApi.v2/Services/ProcessHandler.cs [hunk 12]
```diff
@@ -1,66 +1,82 @@
...
+    {
+        _repository.KillProcess(_mapper.MapFromDTO(proc));
+
     }
 
-    public void ChangePriority(int prio,int pid) {
-        if (prio == 50) { // default priority
+    public void ChangePriority(int prio, int pid)
+    {
+        if (prio == 50)
```
Comment (Horatzio) [inline mockOSApi.v2/Services/ProcessHandler.cs]: Use an Enum for this

File: mockOSApi.v2/Repositories/ProcessRepository.cs (hunk 9 above)
Comment (j0rdan0) [reply mockOSApi.v2/Repositories/ProcessRepository.cs]: I can use this, but  _dbContext.MockProcesses.Where(p => p.pid == pid) returns an IQueryable<MockProcess>, which I don`t really need. Pid needs to be unique, so I only need to get back one entity, which _dbContext.Find<MockProcess> does already. With _dbContext.MockProcesses.Where(p => p.pid == pid) I will need to explicitly cast the IQueryable to a MockProcess?.

File: mockOSApi.v2/Models/Task/MockProcess.cs (hunk 7 above)
Comment (j0rdan0) [reply mockOSApi.v2/Models/Task/MockProcess.cs:56]: done

File: mockOSApi.v2/Data/DTO/ProcessDto.cs (hunk 2 above)
Comment (j0rdan0) [reply mockOSApi.v2/Data/DTO/ProcessDto.cs:9]: Will be an int value, not specified value

File: mockOSApi.v2/Data/DTO/ProcessDto.cs (hunk 3 above)
Comment (j0rdan0) [reply mockOSApi.v2/Data/DTO/ProcessDto.cs:7]: removed this, will refactor

File: mockOSApi.v2/Data/DTO/ProcessDto.cs (hunk 4 above)
Comment (j0rdan0) [reply mockOSApi.v2/Data/DTO/ProcessDto.cs]: Have removed manually created mappers and used Automapper instead, thank for tip ;)

File: mockOSApi.v2/Models/OSObject.cs (hunk 5 above)
Comment (j0rdan0) [reply mockOSApi.v2/Models/OSObject.cs]: removed this

File: mockOSApi.v2/Models/OSObject.cs (hunk 6 above)
Comment (j0rdan0) [reply mockOSApi.v2/Models/OSObject.cs]: removed SetHashCode as will not need it in future

File: mockOSApi.v2/Models/Task/MockProcess.cs (hunk 8 above)
Comment (j0rdan0) [reply mockOSApi.v2/Models/Task/MockProcess.cs]: got it

File: mockOSApi.v2/Services/ProcessHandler.cs (hunk 11 above)
Comment (j0rdan0) [reply mockOSApi.v2/Services/ProcessHandler.cs]: renamed

File: mockOSApi.v2/Services/ProcessHandler.cs (hunk 12 above)
Comment (j0rdan0) [reply mockOSApi.v2/Services/ProcessHandler.cs]: added a const for DEFAULT_PRIORITY since in rest of code the prio can take arbitrary int values

**Comment by j0rdan0** [comment]
DTO layer added, merging into main

**Pull request merged**